### ISSUE TITLE
Inconsistent use of the variable ''vs'' in component DynamicPipe when nParallel > 1

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#247]
**Reported by anton&#x2e;sodja on 27 Nov 2009 13:34 UTC**
The variable _vs_ is described as 'mean velocities in flow segments' in the component _Fluid.Pipes.BaseClasses.PartialTwoPortFlow_ and calculated as follows:
```
SI.Velocity[n] vs = {0.5*(m_flows[i] + m_flows[i+1])/mediums[i].d/crossAreas[i] for i in 1:n}
```
The parmeter _nParallel_ is not taken into account. That would be right if crossAreas are sum of cross areas of all pipes in parallel.
However, the _vs_ variable if further propagated to _FlowModel_ and _HeatTransfer_, nested submodels of _DynamicPipe_, where it is used to calculate Reynolds number as follows:
```
  Res = CharacteristicNumbers.ReynoldsNumber(vs/nParallel, ds, mus, diameters);
```
(note the vs/nParallel argument), but (for example):
```
 if use_Ib_flows then
    Ib_flows = {rhos[i]*vs[i]*vs[i]*crossAreas[i] - rhos[i+1]*vs[i+1]*vs[i+1]*crossAreas[i+1] for i in 1:n-1};
```
Parameter 'crossArea' is ostensibly total cross area at the boundary, although from the Properties dialog of the 'DynamicPipe' component it is not exactly clear what it is and also how should parameter _perimeter_ be set. The hydraulic diameter is calculated as follows:
```
  extends BaseClasses.PartialTwoPortFlow(
    /* ... */
    final crossAreas=fill(crossArea, n),
    final dimensions=fill(4*crossArea/perimeter, n),
```
If crossArea is the total cross area of all parallel pipes and perimeter is the sum of perimeters of all parallel pipes, why then even have parameter _nParallel_?

To sum up, a lot of geometric parameters are confusing when _nParallel_ is set to more then 1 and furthermore when varable _vs_ corresponds to mean flow velocity, then Reynolds number is possibliy calculated wrongly (and consequently heat transfer coefficient).

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/247

**Comment by modelica-trac-importer** [comment]
**Comment by rfranke on 10 Dec 2009 06:12 UTC**
Please see the current SVN version of Modelica. Doesn't the fix introduced on November 13/16 (e011b1b5321e9616408cfee64d7ff66cc53cc55e) solve the reported problem?

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 10 Dec 2009 07:04 UTC**
Unfortunately the above link will not work if you are not logged in (it's kind of a result of our subversion structure). Here is the same [changeset for not-logged-in-users](http://trac.modelica.org/Modelica/changeset/3131/Modelica).

**Comment by modelica-trac-importer** [comment]
**Comment by rfranke on 18 Dec 2012 18:26 UTC**
Having worked on this for fixing #437 today, I'm pretty sure that the problem basically had been solved with e011b1b5321e9616408cfee64d7ff66cc53cc55e. In particular nParallel is considered when obtaining the velocities vs since then. A minor additional fix was introduced today in 1501a366bd304ea394f956f985155ffd49c9704f for Ib_flows.

The geometry paramters (like roughness or crossArea) describe one physical pipe out of nParallel equal pipes. The total cross area is nParallel\*crossArea.

The nominal values of the flow models (like dp_nominal, m_flow_nominal, m_flow_small) relate to the total pipe model.

**Comment by modelica-trac-importer** [comment]
**Changelog removed by rfranke on 18 Dec 2012 18:26 UTC**

**Comment by modelica-trac-importer** [comment]
**Modified by rfranke on 18 Dec 2012 18:26 UTC**

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 20 Dec 2012 08:17 UTC**
